docs/4.2/gcp_guide.md

```diff
@@ -0,0 +1,203 @@
+    project_id: gcp-proj-with-firestore-enabled
+    credentials_path: /var/lib/teleport/firestore_creds
+```
+
+### Storage: Google Cloud Storage
+
+
+When creating the Bucket we would recommend setting it up as `Dual-region` and with 
+`Standard` storage class. Provide access using a `Uniform` access control with a Google-managed
+key. 
+
+When setting up `audit_session_uri` use `gs://` session prefix.
+
+```
+storage:
+    ...
+    audit_sessions_uri: 'gs://teleport-session-storage-2?credentialsPath=/var/lib/teleport/gcs_creds&projectID=bc-jdurbin'
+    ...
+```
+
+### Network Services: Load Balancing
+
+Load Balancing is required for Proxy and SSH traffic `TCP Load Balancing` as Teleport requires
+custom ports for SSH and Web Traffic.
+
+### Network Services: Cloud DNS
+
+Cloud DNS is used to setup the public URL of the Teleport Proxy. Once setup an `A` 
+record is sufficient. 
+
+### Access: Service accounts 
+
+The Authentication server will need to read and write to Firestore for this it'll need to have
+the correct permission via Server Accounts. Learn how to [enable and create service accounts for instances](https://cloud.google.com/compute/docs/access/create-enable-service-accounts-for-instances).
+
+
+## GCP Quickstart
+
+## 1. Create Resources
+We recommend starting by creating the resources. We highly recommend creating these
+an infrastructure automation tool such as [Cloud Deployment Manager](https://cloud.google.com/deployment-manager/) or Terraform. 
+
+ 
+## 2. Install & Configure Teleport
+
+Follow install instructions from our [installation page](installation.md#linux). 
+ 
+We recommend configuring the Teleport in the below steps. 
+
+**1. Configure Teleport Auth Server** using the below example `teleport.yaml`, and start it using SystemD
+
+```yaml
+#
+# Sample Teleport configuration teleport.ymal file for Auth Server
+#
+teleport:
+  nodename: C02WG09CHTDH
+  data_dir: /var/lib/teleport
+  pid_file: /var/run/teleport.pid
+  auth_token: cluster-join-token
+  auth_servers:
+  - 0.0.0.0:3025
+  connection_limits:
+    max_connections: 15000
+    max_users: 250
+  log:
+    output: stderr
+    severity: DEBUG
+  storage:
+    type: firestore
+    collection_name: cluster-data
+    credentials_path: /var/lib/teleport/gcs_creds
+    project_id: example_Teleport-Project-Name
+    audit_events_uri: 'firestore://events?projectID=example_Teleport-Project-Name&credentialsPath=/var/lib/teleport/gcs_creds'
+    audit_sessions_uri: 'gs://teleport-session-storage-2?credentialsPath=/var/lib/teleport/gcs_creds&projectID=example_Teleport-Project-Name'
+auth_service:
+  enabled: yes
+  listen_addr: 0.0.0.0:3025
+  client_idle_timeout: 0s
+  disconnect_expired_cert: false
+  keep_alive_count_max: 0
+```
+
+**2. Setup Proxy**
+
+Save the following configuration file as `/etc/teleport.yaml` on the Proxy Server:
+
+```yaml
+teleport:
+  auth_token: cluster-join-token
+  # you can also use auth server's IP, i.e. "10.1.1.10:3025"
```

Comment:
It's not recommended to use IPs in HA; people should use a TCP load balancer pointed to the auth servers